mantidproject/mslice
# Added ADSObserver to MSlice

**Description of work:**
So far MSlice interacted with the ADS in a one-sided way. Some of its workspaces were stored in the ADS and occasionally deleted, but changes to the ADS from the Mantid side ignored. This sometimes led to missing workspaces in MSlice when the respective workspace was deleted on the Mantid side. However, the respective workspace would still get displayed in MSlice and cause a crash when selected.

This PR introduces an ADS observer that modifies MSlice workspaces on delete, clear and rename, respectively. 
In addition, two small bugs were fixed that became obvious when testing these changes:
1. When taking an interactive cut and closing the interactive cut window without saving the cut to the workspace, a cut was showing up in the MD Histo tab. 
2. When clearing all workspace, there were still workspaces shown in the MD Histo tab that were not available anymore and caused crashes on selection. This was particularly visible after a clear() from the ADS that should cause all workspaces in all tabs to get cleared.

**To test:**

With workspaces saved to the ADS from the MSlice interface, check that the ADS commands clear, delete and rename work as expected in MSlice when run from Mantid.


Fixes #839.
